mewarpx/examples/capacitive_discharge.py

```diff
@@ -243,43 +228,33 @@
 # simulation run
 ##########################
 
-import matplotlib.pyplot as plt
-
-#my_solver = PoissonSolverPseudo1D(nx, ny, D_CA / nx)
+sim.step()
 
-def direct_solve():
-    rho_data = mwxrun.get_rho_grid()[0]
-    rho = rho_data[:,:,0].T
-    phi = my_solver.solve(rho,
-        VOLTAGE * np.sin(2.0*np.pi*FREQ* (mwxrun.get_it()-1.0)*mwxrun.get_dt())
-    )
-    mwxrun.set_phi_grid(phi)
-
-def plot_phi():
-    phi_data = mwxrun.get_gathered_phi_grid()
-    # phi_data = mwxrun.get_phi_grid()
-    if mwxrun.me == 0:
-        print(phi_data)
-        phi_data = phi_data[0]
-        print(mwxrun.me, phi_data.shape)
-        plt.plot(np.mean(phi_data[1:-1], axis=1), 'o-')
-    # plt.plot(phi_data[:,4], 'o-')
-    # plt.plot(phi[:,4], 'o-')
-
-# comment line below to use the multigrid solver
-# callbacks.installfieldsolver(direct_solve)
-
-callbacks.installafterstep(plot_phi)
+##########################
+# collect diagnostics
+##########################
 
-sim.step(5)
 if mwxrun.me == 0:
-    plt.ylim(-2, 28)
-    plt.xlabel('Cell number')
-    plt.ylabel('$\phi$ (eV)')
-    plt.title(
-        'Electrostatic potential at different times gathered\n'
-        'to the root proc from a 2 proc simulation'
-    )
-    plt.grid()
-    plt.savefig('gathered_phi.png')
-    plt.show()
+    import glob
+    import yt
+
+    data_dirs = glob.glob('diags/diags*')
```

Comment:
I would check that data_dirs is not empty before proceeding, as otherwise rho_data division on line 259 will fail since that var was never created, which is a weird error when the problem is accessing data dirs.